### PR TITLE
Fix/pattern navigation

### PR DESCRIPTION
**Pull request by mbeisel**
Fully test graph functionality (a lot of bugs got fixed and features added)
Added edit icon/name (requires backend commit for namechange)
MAP Import support
Fixed add new pattern issue, where the markdown wouldnt get loaded
Check readthedocs

### CODE REVIEW

File: src/app/pattern-language-management/create-pattern/create-pattern.component.ts [hunk 1]
```diff
@@ -192,13 +200,18 @@ export class CreatePatternComponent implements OnInit {
       if (sectionIndex !== -1) {
         const sectionContent = [];
         for (let i = sectionIndex + 1; i < lines.length; i++) {
+          console.log(lines[ i ])
```
Comment (manuwei) [inline src/app/pattern-language-management/create-pattern/create-pattern.component.ts:203]: do we still need this log message?

File: src/app/pattern-language-management/pattern-language-management/pattern-language-management.component.ts [hunk 2]
```diff
@@ -38,75 +37,75 @@ import { UriConverter } from '../../core/util/uri-converter';
...
-    }
+  ngOnInit() {
+    this.patternLanguages = Array.from<PatternLanguageModel>(this.activatedRoute.snapshot.data.patternlanguages.values())
+      .sort(PatternLanguageManagementComponent.sortPatternLanguages);
+  }
 
-    navigateToPL(pl: UriEntity): void {
-      this.zone.run(() => {
-        this.router.navigate([UriConverter.doubleEncodeUri(pl.uri)], { relativeTo: this.activatedRoute });
+  // reload the current data from https://purl.org/patternpedia that contains all patternlangauges
```
Comment (manuwei) [inline src/app/pattern-language-management/pattern-language-management/pattern-language-management.component.ts:69]: I think this comment is really really old since we switched to the database instead of this webside approx. 1 year ago? :D :D

File: src/app/pattern-view-management/pattern-view-renderer/pattern-view-renderer.component.ts [hunk 3]
```diff
@@ -174,12 +176,87 @@ export class PatternViewRendererComponent implements OnInit, AfterViewInit {
...
+        if (dialogResult !== undefined && dialogResult.deleteLink === undefined) { //edit edge
+          this.deleteEdge(this.graphDisplayComponent.currentEdge);
+          this.linkAddedInGraphEditor(dialogResult);
+        } else if (dialogResult !== undefined && dialogResult.deleteLink === true) { // delete Edge
+          this.deleteEdge(edge);
+        } else { //abort
+        }
       });
+    })
+    // this.patternViewService.removeLinksFromView(this.patternViewResponse, edge);
```
Comment (manuwei) [inline src/app/pattern-view-management/pattern-view-renderer/pattern-view-renderer.component.ts:244]: can the comments be deleted?

**Review by manuwei — COMMENTED** [review]
Looks good to me! I just added a few minor comments, mostly to improve  overall code readability (not necessary written by you :) )

File: src/app/core/default-pattern-renderer/default-pattern-renderer.component.html [hunk 4]
```diff
@@ -18,12 +19,12 @@
       <div *ngIf="directedPatternRelations?.length > 0">
         <ng-container *ngFor="let relation of directedPatternRelations">
           <p class="horiz-centered">
-            <a [routerLink]="['..',relation.sourcePatternId]">{{ relation?.sourcePatternName}}</a>
+            <a [routerLink]="['..',relation.sourcePatternUri]">{{ relation?.sourcePatternName}}</a>
```
Comment (manuwei) [inline src/app/core/default-pattern-renderer/default-pattern-renderer.component.html:22]: is this change intended?

File: src/app/core/default-pattern-renderer/default-pattern-renderer.component.ts [hunk 5]
```diff
@@ -89,7 +89,7 @@ export class DefaultPatternRendererComponent implements AfterViewInit, OnDestroy
     const toPattern = edge instanceof DirectedEdgeModel ? edge.targetPatternId : edge.pattern2Id;
     return this.patternService.getPatternContentByPattern(this.patterns.find(it => it.id === toPattern)).pipe(
       map((patterncontent) => {
-        const targetPatternContent = patterncontent.content;
+        // const targetPatternContent = patterncontent.content;
```
Comment (manuwei) [inline src/app/core/default-pattern-renderer/default-pattern-renderer.component.ts:92]: can we delete these two lines ?

Comment (manuwei) [reply src/app/core/default-pattern-renderer/default-pattern-renderer.component.ts:92]: actually we could delete the whole mapping? ( map((pattercontent)...

File: src/app/core/default-pattern-renderer/default-pattern-renderer.component.ts [hunk 6]
```diff
@@ -125,7 +125,9 @@ export class DefaultPatternRendererComponent implements AfterViewInit, OnDestroy
   getPatternLanguageLinks(): Observable<any> {
     const $getDirectedEdges = this.getDirectedEdges();
     const $getUndirectedEdges = this.getUndirectedEdges();
-    return forkJoin([$getDirectedEdges, $getUndirectedEdges]).pipe(tap(() => this.isLoadingLinks = false));
+    console.log($getDirectedEdges)
```
Comment (manuwei) [inline src/app/core/default-pattern-renderer/default-pattern-renderer.component.ts:128]: should be deleted

File: src/app/core/default-pattern-renderer/default-pattern-renderer.component.ts [hunk 7]
```diff
@@ -207,22 +209,33 @@ export class DefaultPatternRendererComponent implements AfterViewInit, OnDestroy
...
     } else {
       this.patternLanguageService.getPatternLanguageByEncodedUri(this.patternLanguageId).pipe(
         tap((patternLanguage) => this.patternLanguage = patternLanguage),
         // get our individual pattern and the links in parallel
-        switchMap(() => forkJoin([this.fillPatternSectionData(), this.getPatternLanguageLinks()])));
+        switchMap(() => forkJoin([ this.fillPatternSectionData(), this.getPatternLanguageLinks() ])));
     }
-    const dataSubscription = dataObservable.subscribe(() =>
-      this.cdr.detectChanges());
+    const dataSubscription = getPatternObservable.subscribe(() => dataObservable.subscribe(() =>
```
Comment (manuwei) [inline src/app/core/default-pattern-renderer/default-pattern-renderer.component.ts:233]: we should pipe those requests instead of subscribing in a subscribe...

**Review by manuwei — APPROVED** [review]
The PR looks fine to me :)  I added a few minor comments, you can resolve them and merge afterwards

File: src/app/core/default-pattern-renderer/default-pattern-renderer.component.html (hunk 4 above)
Comment (mbeisel) [reply src/app/core/default-pattern-renderer/default-pattern-renderer.component.html:22]: yes it was the reason for the error when routing via patternlinks and accessing the pattern via the info button in the graph.
Instead of accessing the pattern via: patternpedia/PL/Patternuri patternpedia/PL/P_ID got called - which does not exist

**Review by manuwei — APPROVED** [review]
